test/src/tests/account.test.js

```diff
@@ -5,18 +5,14 @@ import {
   vListItem,
   vIcon,
 } from 'jest-puppeteer-vuetify';
-import { CLIENT_URL, uniqueId } from '../util';
+import { uniqueId, registerNewUser } from '../util';
 
 describe('account management', () => {
-  const username = `user${uniqueId()}`;
-  const email = `${username}@dandi.test`;
-  const password = 'password'; // Top secret
-
-  beforeAll(async () => {
-    await page.goto(CLIENT_URL);
-  });
-
   it('registers a new user', async () => {
+    const username = `user${uniqueId()}`;
+    const email = `${username}@dandi.test`;
+    const password = 'password'; // Top secret
+
```

Comment:
It seems like much of this test overlaps with the `registerNewUser` utility function. Any reason not be using that here?

Reply:
This is a testing methodology thing. My thinking is that helpers and utilities should be treated like black boxes which magically change the app state. Tests need to explicit about what they're testing, so they should never be using helpers or utilities to perform the actions being tested. Helpers should only be used to set up or tear down test data.
In this case, the helper and the test code are basically the same, but that won't necessarily be the case for other helpers, and it's possible that the helper will be changed in the future. Helpers should be free to take whatever shortcuts they can to speed up test set up, and no tests should break if a helper changes (assuming the helper still does what it is advertised to do).
The code duplication is sad, but IMHO it's necessary.

Reply:
Agreed!